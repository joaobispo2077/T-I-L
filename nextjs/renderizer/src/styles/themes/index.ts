export const themes = {
  default: {
    colors: {
      primary: '#0A1128',
      secondary: '#dc143C',
      light: '#FFFFFF',
      dark: '#0A1128',
      mediumGray: '#DDDDDD',
    },
    font: {
      family: {
        primary: "'Open Sans', sans-serif",
        secondary: "'Montserrat', sans-serif",
      },
      sizes: {
        xsmall: '.8rem',
        small: '1.6rem',
        medium: '2.4rem',
        large: '3.2rem',
        xlarge: '4.0rem',
        xxlarge: '4.8rem',
        huge: '5.6rem',
        xhuge: '6.4rem',
      },
    },
    spacings: {
      xsmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
    media: {
      lteMedium: '(max-width: 768px)',
    },
  },
} as const;
