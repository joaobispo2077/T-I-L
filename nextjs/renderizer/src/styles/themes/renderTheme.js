import { ThemeProvider } from 'styled-components';
import { themes } from '.';

export const renderTheme = (children) => {
	return (
		<ThemeProvider theme={{ ...themes['default'] }}>{children}</ThemeProvider>
	);
};
