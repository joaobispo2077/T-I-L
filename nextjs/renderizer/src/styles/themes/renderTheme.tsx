import React from 'react';
import { ThemeProvider } from 'styled-components';
import { themes } from '.';

export const renderTheme = (children: React.ReactNode) => {
  return (
    <ThemeProvider theme={{ ...themes.default }}>{children}</ThemeProvider>
  );
};
