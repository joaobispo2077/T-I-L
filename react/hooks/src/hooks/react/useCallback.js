import React, { useCallback, useState } from 'react';
import PropTypes from 'prop-types';
import './App.css';

// store component into memory to not recreate if not necessary
const Button = React.memo(function Button({ incrementCounter }) {
  console.log('render button');
  return <button onClick={() => incrementCounter(10)}>+</button>;
});

Button.propTypes = {
  incrementCounter: PropTypes.func,
};

function App() {
  const [counter, setCounter] = useState(0);

  // store ***function*** into memory to not recreate when component render
  const handleIncrementCounter = useCallback((number) => {
    setCounter((prevCounter) => prevCounter + number);
  }, []);

  console.log('Render app');
  return (
    <div className="App">
      <h1>Contadores</h1>

      <h2>Counter: {counter}</h2>

      <Button incrementCounter={handleIncrementCounter} />
    </div>
  );
}

export default App;
