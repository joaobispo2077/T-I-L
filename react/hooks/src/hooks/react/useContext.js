import './App.css';
import React from 'react';
import { Box } from './components/Box';
import { AppContext } from './contexts/AppContext';

function App() {
  console.log('Render app');
  return (
    <AppContext>
      <Box />
    </AppContext>
  );
}

export default App;
