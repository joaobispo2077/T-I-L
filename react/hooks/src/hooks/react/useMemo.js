import React, { useEffect, useMemo, useState } from 'react';
import PropTypes from 'prop-types';
import './App.css';

const Post = ({ post }) => {
  console.log('render post');
  return (
    <div className="post">
      <h1>{post.title}</h1>
      <p>{post.body}</p>
    </div>
  );
};

Post.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    body: PropTypes.string,
  }),
};

const Posts = ({ posts = [] }) => {
  return (
    <>
      {posts.map((post) => (
        <Post key={post.id} post={post} />
      ))}
    </>
  );
};

Posts.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      body: PropTypes.string,
    }),
  ),
};

function App() {
  const [posts, setPosts] = useState([]);
  const [value, setValue] = useState('');
  console.log('Render app');

  useEffect(() => {
    setTimeout(() => {
      fetch('https://jsonplaceholder.typicode.com/posts')
        .then((response) => response.json())
        .then((posts) => setPosts(posts));
    }, 1000);
  }, []);

  const MemorizedPosts = useMemo(() => <Posts posts={posts} />, [posts]);

  return (
    <div className="App">
      <p>
        <input
          type="search"
          value={value}
          onChange={(event) => setValue(event.target.value)}
        />
      </p>

      {MemorizedPosts}

      {posts.length <= 0 && <p>Ainda n√£o carregaram os posts</p>}
    </div>
  );
}

export default App;

// simple
// import React, { useCallback, useMemo, useState } from 'react';
// import PropTypes from 'prop-types';
// import './App.css';

// const Button = ({ incrementCounter }) => {
//   console.log('render button');
//   return <button onClick={() => incrementCounter(10)}>+</button>;
// };

// Button.propTypes = {
//   incrementCounter: PropTypes.func,
// };

// function App() {
//   const [counter, setCounter] = useState(0);

//   const handleIncrementCounter = useCallback((number) => {
//     setCounter((prevCounter) => prevCounter + number);
//   }, []);

//   const MemorizedButton = useMemo(
//     () => <Button incrementCounter={handleIncrementCounter} />,
//     [handleIncrementCounter],
//   );

//   console.log('Render app');
//   return (
//     <div className="App">
//       <h1>Contadores</h1>

//       <h2>Counter: {counter}</h2>

//       {MemorizedButton}
//     </div>
//   );
// }

// export default App;
