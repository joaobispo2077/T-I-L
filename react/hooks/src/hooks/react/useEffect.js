import { useState } from 'react';
import { useEffect } from 'react';
import './App.css';

const titleEvent = () => {
  console.log('h1 clicked');
};

function App() {
  const [counter, setCounter] = useState(0);
  const [otherCounter, setOtherCounter] = useState(0);

  // execute every single time that state into component change
  useEffect(() => console.log('ComponentDidUpdate'));

  // execute 1x (when component load for first time)
  useEffect(() => {
    console.log('ComponentDidMount');

    document.querySelector('h1')?.addEventListener('click', titleEvent);

    // componentWillUnmount
    return () => {
      document.querySelector('h1')?.removeEventListener('click', titleEvent);
    };
  }, []);

  // execute when state into array dependencies change
  useEffect(() => {
    console.log(`C1: ${counter}`, `C2: ${otherCounter}`);
  }, [counter, otherCounter]);

  return (
    <div className="App">
      <h1>Contadores</h1>

      <h2>Counter 1: {counter}</h2>
      <h2>Counter 2: {otherCounter}</h2>

      <button onClick={() => setCounter(counter + 1)}>+</button>
      <button onClick={() => setOtherCounter(otherCounter + 1)}>
        + (other counter)
      </button>
    </div>
  );
}

export default App;
