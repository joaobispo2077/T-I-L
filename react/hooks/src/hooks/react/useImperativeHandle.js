/* eslint-disable no-unused-vars */

import { useLayoutEffect, useRef, useState } from 'react';

const Home = () => {
  const [counted, setCounted] = useState([0, 1, 2, 3]);
  const divRef = useRef();

  // this hook is blockant with navigator
  useLayoutEffect(() => {
    divRef.current.scrollTop = divRef.current.scrollHeight;
  });

  const handleIncrementCounted = () => {
    setCounted((prevCounted) =>
      prevCounted.concat(Number(prevCounted.slice(-1)) + 1),
    );
  };

  return (
    <div onClick={handleIncrementCounted}>
      <h1>Home</h1>
      <button>add {counted.slice(-1)}</button>
      <div
        ref={divRef}
        style={{ height: '100px', width: '100px', overflowY: 'scroll' }}
      >
        {Array.isArray(counted) &&
          counted.map((count) => <p key={`c-${count}`}>{count}</p>)}
      </div>
    </div>
  );
};

export default Home;
