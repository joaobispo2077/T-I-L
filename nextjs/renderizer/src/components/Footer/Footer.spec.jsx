import { render, screen } from '@testing-library/react';
import { Footer } from '.';
import { renderTheme } from '../../styles/themes/renderTheme';

describe('<Footer/>', () => {
  it('should render footer', () => {
    render(renderTheme(<Footer html={'<h1>Hello world</h1>'} />));

    expect(screen.getByRole('heading')).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    const { container } = render(
      renderTheme(<Footer html={'<h1>Hello world</h1>'} />),
    );

    expect(container).toMatchSnapshot();
  });
});
