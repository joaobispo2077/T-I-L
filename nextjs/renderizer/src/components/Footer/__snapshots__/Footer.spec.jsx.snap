// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should match with snapshot 1`] = `
.c4 {
  font-size: 2.4rem;
}

.c2 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
  font-size: 1.6rem;
}

.c0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .c3 {
  font-size: 1.6rem;
}

.c0 .c1 {
  padding-top: 0;
  padding-bottom: 0;
}

<div>
  <footer
    class="c0"
  >
    <section
      class="c1 c2"
    >
      <footer
        class="c3 c4"
      >
        <h1>
          Hello world
        </h1>
      </footer>
    </section>
  </footer>
</div>
`;
